example/pilotmode.sw

```diff
@@ -0,0 +1,24 @@
+def cons : a * b -> (a -> b) -> (a -> b) = \p. \k. \a.
+  if (a == fst p) {snd p} {k a}
+end
+
+def nil : a -> cmd unit = \a. return () end
+
+// Suitable to use as e.g.
+//   installKeyHandler "(S-)←↓↑→ [g]rab [h]arvest [d]rill [s]can [u]pload" pilot
+def pilot : key -> cmd unit =
+  cons (key "Up",      move) $
+  cons (key "Down",    turn back) $
+  cons (key "Left",    turn left) $
+  cons (key "Right",   turn right) $
+  cons (key "S-Up",    turn north) $
+  cons (key "S-Down",  turn south) $
+  cons (key "S-Left",  turn west) $
+  cons (key "S-Right", turn east) $
+  cons (key "g",       grab; return ()) $
+  cons (key "h",       harvest; return ()) $
+  cons (key "d",       drill forward; return ()) $
+  cons (key "s",       scan forward; return ()) $
```

Comment:
While trying out this demo, I found that I missed being able to see the result of `scan` and `grab` in the REPL.  But I take it that it is necessary force all commands to have the same return type in the key handler, and therefore suppress the REPL output via `return ()`?

I guess this is a use case to continue to retain the original pilot mode.

Reply:
Yes, that's right.  Though if we implement a simple version of #683, so *e.g.* we have `print : text -> cmd unit` which will output something on the console, then we could use that to print out the results of `grab` and `scan`.

Reply:
Come to think of it, simply using `log` would achieve the same end.

Reply:
Maybe this example should do so?

Reply:
Oh, good idea!  I tried adding logging but it seems to have no effect.  Currently trying to figure out why.